zerver/lib/validator.py

```diff
@@ -440,9 +437,86 @@ def check_string_or_int_list(var_name: str, val: object) -> Optional[str]:
 
     return check_list(check_int)(var_name, val)
 
-@set_type_structure('any(int, str)')
+@set_type_structure((Union, ['str', 'int']))
 def check_string_or_int(var_name: str, val: object) -> Optional[str]:
     if isinstance(val, str) or isinstance(val, int):
         return None
 
     return _('{var_name} is not a string or integer').format(var_name=var_name)
+
+def validate_the_validator(func: FuncT, mypy_type: Any) -> None:
+    if USING_TYPE_STRUCTURE:
+        def check(type_structure: TypeStructure, mypy_type: Any) -> None:
+            if type_structure == 'skip':
+                return
+
+            if isinstance(type_structure, tuple):
+                if type_structure[0] == Union:
+                    print(mypy_type, type_structure)
+                    assert mypy_type.__origin__ in [Union]
+                    return
+
+                if type_structure[0] == List:
+                    assert mypy_type.__origin__ in [Iterable, List, Sequence]
+
+                    sub_type_structure = type_structure[1]
+                    sub_mypy_type = mypy_type.__args__[0]
+                    check(sub_type_structure, sub_mypy_type)
+                    return
+
+                # TODO: validate other complex types
+                return
+
+            if type_structure.startswith('any'):
+                return
+
+            if type_structure == 'bool':
+                assert mypy_type == bool
+                return
+
+            if type_structure == 'int':
+                assert mypy_type == int
+                return
+
+            if type_structure == 'str':
+                assert mypy_type == str
+                return
+
+            print(func.type_structure)
+            print(mypy_type)
+            raise Exception('invalid validator', type_structure)
+
+        check(func.type_structure, mypy_type)
+
+def debug(var_name: str, val: object) -> Optional[str]:
+    # TODO: Remove!
+    print('auto', var_name, val)
+    return None
+
+def auto(var_name: str, val: object) -> Optional[str]:
+    # This is a hack for mypy.  The `auto` function is used
+    # to signify to REQ that we want to use the result
+    # of build_auto_validator for validation.
+    raise Exception('REQ did not detect auto.')
```

Comment:
AssertionError is what you want here.

Reply:
will fix (I did say "please don't merge or review" yet)